util/converter/converter.go

```diff
@@ -124,9 +124,10 @@ func (c *conversion) convert(ctx context.Context, cs content.Store, desc ocispec
 		return nil, err
 	}
 	defer decR.Close()
-	rdr := decR
+	decTR := io.TeeReader(decR, origDiffID.Hash())
+	rdr := decTR
 	if c.rewriteTimestamp != nil {
-		tcR := tarconverter.NewReader(io.TeeReader(decR, origDiffID.Hash()), rewriteTimestampInTarHeader(*c.rewriteTimestamp))
+		tcR := tarconverter.NewReader(decTR, rewriteTimestampInTarHeader(*c.rewriteTimestamp))
```

Comment:
Isn't the issue in `tarconverter` pkg? Iiuc tarconverter reader is returning EOF before it has reached EOF of the input reader? If that is the case then looks like tarconverter returns EOF too early?

Reply:
Updated the PR to fix the tarconverter